test/arrays/promotion/promotedArrayIndexing/const/modify-with-method.good

```diff
@@ -0,0 +1 @@
+modify-with-method.chpl:3: warning: modifying the result of a promoted index expression is unstable
```

Comment:
Huh... Seems like this should generate an error since we're using promotion to modify a const array?  That's independent of the PR but would be worth capturing as an issue, if we haven't already.

Reply:
Yeah its a bug I noticed when I originally deprecated the feature. Bug is https://github.com/chapel-lang/chapel/issues/23215 and it has a future in our tests, `test/arrays/const/modify-const-array-of-record.chpl`

Reply:
I thought it sounded familiar, thanks!  Seems potentially, or at least thematically, related to https://github.com/chapel-lang/chapel/issues/23970